lib/link.js

```diff
@@ -127,6 +127,7 @@ Link.prototype.detach = function() {
 Link.prototype.forceDetach = function() {
   debug('Force-detach for ' + this.name + '. Current state: ' + this.state());
   this.linkSM.forceDetach();
+  this._detached();
```

Comment:
Ok, I've looked through the code and think that this would clean up what needs to clean up, without causing problems:

```
Link.prototype._detached = function(frame) {
  if (this.remoteHandle !== undefined) {
    delete this.session._linksByRemoteHandle[this.remoteHandle];
    this.remoteHandle = undefined;
  }

  this._resolveAttachPromises((frame && frame.error) ? frame.error : 'link closed');

  if (!frame) return; // Cleaned up link on forced detach, ensure we don't try and reattach or send events.

  if (frame && frame.error) {
    this.emit(Link.ErrorReceived, frame.error);
  }

  this.emit(Link.Detached, { closed: frame.closed, error: frame.error });

  var self = this;
  if (!self.shouldReattach()) return;

  if (!self._timeouts.length) self._timeouts = u.generateTimeouts(self.policy.reattach);

  setTimeout(function() {
    self._attemptReattach();
  }, self._timeouts.shift());
};
```